core/src/test/java/io/confluent/rest/JsonMappingExceptionMapperTest.java

```diff
@@ -55,6 +55,22 @@ public void testJsonMappingException() {
     }
   }
 
+  @Test
+  public void testJsonMappingExceptionWithInvalidJson() {
+    try {
+      String json = "{\"name\":{'key:value}}";
+      ObjectMapper mapper = new ObjectMapper();
+      mapper.reader().forType(User.class).readValue(json);
+    } catch (JsonMappingException e) {
+      Response response = mapper.toResponse(e);
+      assertEquals(400, response.getStatus());
+      ErrorMessage out = (ErrorMessage)response.getEntity();
+      assertEquals(400, out.getErrorCode());
```

Comment:
Can you check that the error message is as expected too please, not just the code.